test/pkg/compose/templatingdata/templating_data.go

```diff
@@ -0,0 +1,95 @@
+package templatingdata
+
+import (
+	"fmt"
+	"strings"
+	"text/template"
+
+	"k8s.io/klog/v2"
+)
+
+// TemplatingData contains all values needed for templating Composer's Sources & Blueprints,
+// and other templated artifacts within a MicroShift's test harness.
+type TemplatingData struct {
+	Arch string
+
+	// Minor version of current release's RHOCP. If RHOCP is not available yet, it defaults to 0.
+	RHOCPMinorY int
+
+	// Minor version of previous release's RHOCP. If RHOCP is not available yet, it defaults to 0.
+	RHOCPMinorY1 int
+
+	// Minor version of previous previous release's RHOCP.
+	RHOCPMinorY2 int
```

Comment:
Does the template language support doing math?

Reply:
It's https://pkg.go.dev/text/template. I believe `gomplate` is using that, maybe with minor tweaks. It appears it doesn't support math directly, but we can create and pass function so the template can use them, e.g. https://github.com/openshift/microshift/pull/3169/files#diff-1bb0ae99b6be320a6d429700ab05f47c41c56a05a32695648cebaafea271429cR74-R78
```
	funcs := map[string]any{
		"hasPrefix": strings.HasPrefix,
	}

	tpl, err := template.New(name).Funcs(funcs).Parse(data)
```
so we could have something like this ([go playground](https://go.dev/play/p/femnHkV6ZA-)):
```
type Data struct {
	RHOCPMinor int
}

func main() {
	const tpl = `
RHOCP minor: {{ .RHOCPMinor }}
Minor - 2: {{ add .RHOCPMinor -2 }}`

	data := Data{RHOCPMinor: 16}

	funcs := map[string]any{
		"add": func(num, toAdd int) int { return num + toAdd },
	}

	t := template.Must(template.New("test").Funcs(funcs).Parse(tpl))

	err := t.Execute(os.Stdout, data)
	if err != nil {
		log.Println("executing template:", err)
	}
}
```

Reply:
Maybe it's better to stick with explicit values to avoid syntax issues.